### ISSUE TITLE
Rules of type casting in Model.update are not clear

### ISSUE THREAD

**Issue by dchekanov** [Automattic/mongoose#5403]
In Model.update documentation (http://mongoosejs.com/docs/api.html#model_Model.update) is says:

> All update values are cast to their appropriate SchemaTypes before being sent.

It doesn't work as I expected in this case:

```javascript
var mongoose = require('mongoose');

mongoose.set('debug', true);

var Schema = new mongoose.Schema({
  colors: [{type: String}]
});

var Model = mongoose.model('Test', Schema);

mongoose.connect('mongodb://localhost/db');

mongoose.connection.once('open', function() {
  Model
    .update({}, {colors: 'red'})
    .exec();

  var document = new Model({colors: 'red'});

  document.save();
});
```

```
Mongoose: tests.update({}, { '$set': { colors: 'red' } }, {})
Mongoose: tests.insert({ _id: ObjectId("59515913df3ff64d688abcfb"), colors: [ 'red' ], __v: 0 })
```

As you can see, in the first case it saves a raw value, 'red'. In the second case it converts 'red' to ['red'], which is what I expected to happen with Model.update as well.

Is this how it supposed to work? How do I determine when I need to massage data by myself and when it will be prepared by mongoose?

Node.js 6.9.5
mongoose 4.11.0
MongoDB 3.4.2

**Comment by varunjayaraman** [comment]
fair point, i'll label this an enhancement for `Model#update`: casting non-arrays to an array containing that value as the sole element if the schema type is an array

**Comment by vkarpov15** [comment]
This is a bug in mongoose. The correct behavior is saving an array, and will be fixed for 4.11.2. Thanks for reporting and sorry for the inconvenience.
